holoviews/core/spaces.py

```diff
@@ -542,7 +542,7 @@ def __call__(self, *args, **kwargs):
         # Nothing to do for callbacks that accept no arguments
         kwarg_hash = kwargs.pop('memoization_hash', ())
```

Comment:
Ah, I suppose the `Callable` gets it, but at least the callback does not. Even so, it means `Callable` has kwargs that won't be passed to the callback (if supplied by the user).

Reply:
>How come the hash is passed in via the kwargs? Does that mean that if a user tries to use a stream with the parameter 'memoization_hash', it will not work as expected?

It's the only way we have of letting streams define hash keys to the Callable and has been that way for a while. It is true that if a user stream has a parameter called memoization_hash then that won't work properly. Seems like a highly unlikely clash but if you don't think it's safe I'd be happy to add an underscore or something.

>Even so, it means Callable has kwargs that won't be passed to the callback (if supplied by the user).

Don't quite follow this, when or why would a user call a `Callable`?

Reply:
> Don't quite follow this, when or why would a user call the Callable

I don't expect this to be a common case, but ideally the `Callable` is a relatively transparent wrapper around the callback. This isn't a blocker for this PR though...